tests/testthat/test-correlation.R

```diff
@@ -2,104 +2,84 @@ test_that('1 correlation between rounds gives sensible samples', {
   round_2 <- sample_intervention(target, 'pev', vaccine_coverage, correlations)
-  expect_equal(
-    length(intersect(which(round_1), which(round_2))),
-    pop * .5,
-    tolerance=1e2
-  )
-  expect_equal(sum(round_1), sum(round_2), tolerance=1e2)
-  expect_equal(sum(round_1), pop * .5, tolerance=1e2)
+
+  expect_equal(sum(round_1), sum(round_2), tolerance=.1)
+  expect_equal(sum(round_1), pop * .5, tolerance=.1)
+  expect_equal(sum(round_1 & round_2), pop * .25, tolerance=.1)
+  expect_equal(sum(round_1 | round_2), pop * .75, tolerance=.1)
 })
 
 test_that('1 correlation between interventions gives sensible samples', {
   pop <- 1e6
   target <- seq(pop)
   vaccine_coverage <- .2
   mda_coverage <- .2
-  parameters <- get_parameters(list(
-    human_population = pop,
-    pev = TRUE,
-    mda = TRUE
-  ))
-  correlations <- get_correlation_parameters(parameters)
+  correlations <- CorrelationParameters$new(pop, c('pev', 'mda'))
   correlations$inter_round_rho('pev', 1)
   correlations$inter_round_rho('mda', 1)
   correlations$inter_intervention_rho('pev', 'mda', 1)
   vaccine_sample <- sample_intervention(target, 'pev', vaccine_coverage, correlations)
   mda_sample <- sample_intervention(target, 'mda', mda_coverage, correlations)
 
-  expect_equal(sum(vaccine_sample), pop * .2, tolerance=1e2)
-  expect_equal(sum(mda_sample), pop * .2, tolerance=1e2)
-  expect_equal(sum(vaccine_sample & mda_sample), pop * .2, tolerance=1e2)
+  expect_equal(sum(vaccine_sample), pop * .2, tolerance=.1)
+  expect_equal(sum(mda_sample), pop * .2, tolerance=.1)
+  expect_equal(sum(vaccine_sample & mda_sample), pop * .2, tolerance=.1)
+  expect_equal(sum(vaccine_sample | mda_sample), pop * .2, tolerance=.1)
 })
 
 test_that('0 correlation between interventions gives sensible samples', {
   pop <- 1e6
   target <- seq(pop)
   vaccine_coverage <- .2
   mda_coverage <- .2
-  parameters <- get_parameters(list(
-    human_population = pop,
-    pev = TRUE,
-    mda = TRUE
-  ))
-  correlations <- get_correlation_parameters(parameters)
+  correlations <- CorrelationParameters$new(pop, c('pev', 'mda'))
   correlations$inter_round_rho('pev', 1)
   correlations$inter_round_rho('mda', 1)
   correlations$inter_intervention_rho('pev', 'mda', 0)
+
   vaccine_sample <- sample_intervention(target, 'pev', vaccine_coverage, correlations)
   mda_sample <- sample_intervention(target, 'mda', mda_coverage, correlations)
-  expect_equal(
-    length(intersect(which(vaccine_sample), which(mda_sample))),
-    pop * .5,
-    tolerance=1e2
-  )
-  expect_equal(sum(vaccine_sample), sum(mda_sample), tolerance=1e2)
-  expect_equal(sum(vaccine_sample), pop * .5, tolerance=1e2)
+
+  expect_equal(sum(vaccine_sample), sum(mda_sample), tolerance=.1)
+  expect_equal(sum(vaccine_sample), pop * .2, tolerance=.1)
+  expect_equal(sum(vaccine_sample & mda_sample), pop * .04, tolerance=.1)
+  expect_equal(sum(vaccine_sample | mda_sample), pop * .36, tolerance=.1)
```

Comment:
Where do these population proportions come from? I would expect .1 and .3

Perhaps it would be clearer to replace vaccine_coverage and mda_coverage with `coverage <- .2`. And then test

```suggestion
  expect_equal(sum(vaccine_sample & mda_sample), pop * coverage * .5, tolerance=.1)
  expect_equal(sum(vaccine_sample | mda_sample), pop * coverage * 1.5, tolerance=.1)
```

Reply:
With a correlation of zero, these are independent events so the probability of being in both is `vaccine_coverage * mda_coverage`, ie. 0.04.

The probability of being in either/or is `vaccine + mda - (vaccine * mda)`, or `0.2 + 0.2 - 0.04 = 0.36`.

I agree having the explicit calculation is clearer, I'll change to that.

Reply:
Added the explicit calculations.

I actually decided to go the opposite direction, keeping the separate variables for pev and mda but using different values for each of them. This feels like a more meaningful test. I've renamed vaccine_xx with pev_xx to match the name of the intervention passed as a string.